src/utils/semver.js

```diff
@@ -47,3 +54,15 @@ export async function getCleanDependencyName([depName, range]) {
 
   return [`${depName}@${range}`, `${depName}@${depVer}`, isLatest];
 }
+
+export function addSASTWarning(dirPath) {
+  const packageJsonPath = path.join(dirPath, "package.json");
+
+  const packageJsonVersion = JSON.parse(readFileSync(packageJsonPath, "utf8")).version;
+
+  if (KVersionsToWarn.includes(packageJsonVersion)) {
+    throw new Error("The version of package.json is invalid");
```

Comment:
Hi @PierreDemailly , I admit I didn't quite understand that suggestion.

Reply:
Hi @fabnguess 
Nevermind, my bad: I miss understood your function. You can finish this issue by adding the warning, probably here:
```js
const fileAnalysisRaw = await Promise.allSettled(
  files
  .filter((name) => kJsExtname.has(path.extname(name)))
  .map((file) => scanJavascriptFile(dest, file, name))
);

const fileAnalysisResults = fileAnalysisRaw
  .filter((promiseSettledResult) => promiseSettledResult.status === "fulfilled")
  .map((promiseSettledResult) => promiseSettledResult.value);

ref.warnings.push(...fileAnalysisResults.flatMap((row) => row.warnings));
```

You'll still have to edit your function, actually it would mean something like:
```js
try {
   addSASTWarning(...);
} catch {
  // add the warning
} 
```
It's weird, the function name is wrong I think, shoulda be something like
```js
try {
   verifySemVer(...);
} catch {
  // add the warning
} 
```
About the `try-catch`, it's even possible to return null if the SemVer is valid.
```js
const error = verifySemVer(...); // or checkSemVer or checkPackageVersion or anything...
if (error) {
  // add the warning
}
```
Or you can even use `@openally/result` if you want to keep the `try-catch` ^^
I let you see :)